sdk/src/transaction/mod.rs

```diff
@@ -452,6 +453,16 @@ impl Transaction {
                     libsecp256k1_fail_on_bad_count,
                 );
                 e.map_err(|_| TransactionError::InvalidAccountIndex)?;
+            } else if crate::ed25519_program::check_id(program_id) {
```

Comment:
There's a similar consensus issue here as well.  Since `FeatureSet` is not available, we'll need new parameter into `verify_precompiles()` that indicates whether  `ed25519_program` has been enabled yet or not.  

It might be better to just actually pass `FeatureSet` into `verify_precompiles()`, perhaps as a separate PR that lands before this one.  More context: soon after this PR lands I'll likely be adding another precompile program in here too!

Reply:
I've added another commit to do exactly this.

Reply:
The commits should really be re-ordered.